From 46e0af2942e23fb3cf1c313e58e4081877d4f289 Mon Sep 17 00:00:00 2001
From: Dan Williams <dcbw@redhat.com>
Date: Thu, 30 Aug 2012 13:57:06 -0500
Subject: [PATCH] ip6: ignore cached/cloned routes when retrieving IPv6 config
 (lp:1038541) (bgo #671767)

Should also ignore them when building up the initial IPv6 config.
---
 src/ip6-manager/nm-ip6-manager.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/ip6-manager/nm-ip6-manager.c b/src/ip6-manager/nm-ip6-manager.c
index 3dc8616..4432aa5 100644
--- a/src/ip6-manager/nm-ip6-manager.c
+++ b/src/ip6-manager/nm-ip6-manager.c
@@ -1531,6 +1531,12 @@ nm_ip6_manager_get_ip6_config (NMIP6Manager *manager, int ifindex)
 		if (rtnl_route_get_family (rtnlroute) != AF_INET6)
 			continue;
 
+		/* And ignore cache/cloned routes as they aren't part of the interface's
+		 * permanent routing configuration.
+		 */
+		if (rtnl_route_get_flags (rtnlroute) & RTM_F_CLONED)
+			continue;
+
 		nldest = rtnl_route_get_dst (rtnlroute);
 		if (!nldest || nl_addr_get_family (nldest) != AF_INET6)
 			continue;
-- 
1.7.12

